pycc/cclambda.py

```diff
@@ -129,8 +129,14 @@ def solve_lambda(self, e_conv=1e-7, r_conv=1e-7, maxiter=100, max_diis=8, start_
             Wabef_cc3 = self.build_cc3_Wabef(o, v, ERI, t1)
 
             # Building intermediates in t3l1
-            Zmndi = np.zeros((no,no,nv,no))
-            Zmdfa = np.zeros((no,nv,nv,nv))
+            if isinstance(t1, torch.Tensor):                
+                Zmndi = torch.zeros_like(t2[:,:,:,:no])
+                Zmdfa = torch.zeros_like(t2)
```

Comment:
Using np.zeros_like()/torch.zeros_like() to keep the new intermediates to be the same type and on the same device as the amplitudes. No need to specify the type for different settings (sp/dp, CPU/GPU).